gossip-itest/src/test/java/org/apache/gossip/PerNodeDataEventTest.java

```diff
@@ -0,0 +1,136 @@
+/*
```

Comment:
Do more unit testing. The end to end testing is good but refactor the code in a way that you can test if events are being fired